webapp/src/components/jira_field.jsx

```diff
@@ -62,29 +57,31 @@ export default class JiraField extends React.PureComponent {
         if (field.schema.type === 'string') {
             return (
                 <Input
-                    key={field.key}
-                    id={field.key}
+                    key={fieldKey}
+                    id={fieldKey}
                     label={field.name}
                     type='input'
-                    onChange={this.handleChange}
-                    required={this.props.obeyRequired && field.required}
+                    onChange={this.props.onChange}
+                    required={obeyRequired && field.required}
                     value={this.props.value}
                 />
             );
         }
 
+        // if this.props.field has allowedValues, then props.value will be an object
         if (field.allowedValues && field.allowedValues.length) {
             const options = field.allowedValues.map(this.makeReactSelectValue);
+
             return (
                 <ReactSelectSetting
-                    key={field.key}
-                    name={field.key}
+                    key={fieldKey}
+                    name={fieldKey}
                     label={field.name}
                     options={options}
-                    required={this.props.obeyRequired && field.required}
-                    onChange={this.handleChange}
+                    required={obeyRequired && field.required}
+                    onChange={(id, val) => this.props.onChange(id, {id: val})}
                     isMulti={false}
-                    value={options.filter((option) => option.value === this.props.value)}
+                    value={options.find((option) => option.value === this.props.value)}
```

Comment:
A couple of notes here: 

Nice catch here to use `.find()` to return the object instead of `.filter()` which returns an array.
Also nice choice to assign onChange here so that `handleFieldChange` does not have to handle values of string and objects.